tests/nonroot_test.go

```diff
@@ -61,4 +62,29 @@ var _ = Describe("[sig-compute]NonRoot feature", func() {
 		table.Entry("SRIOV", sriovVM, "", "SRIOV"),
 		table.Entry("VirtioFS", virtioFsVM, virtconfig.VirtIOFSGate, "VirtioFS"),
 	)
+
+	Context("[verify-nonroot] NonRoot feature", func() {
+		It("Fails if can't be tested", func() {
+			Expect(checks.HasFeature(virtconfig.NonRoot)).To(BeTrue())
+
+			vmi := tests.NewRandomVMI()
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, util.NamespaceTestDefault)
+			podOutput, err := tests.ExecuteCommandOnPod(
+				virtClient,
+				vmiPod,
+				vmiPod.Spec.Containers[0].Name,
+				[]string{"id"},
+			)
+
+			groups := strings.Split(podOutput, "=")
+			uid := strings.Split(groups[1], "(")[0]
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(uid).To(Equal("107"))
+		})
```

Comment:
is this better than simply ensuring the security context for the pod and each container is nonRoot on the pod spec itself?

Reply:
We already have a test that does it. I think this doesn't hurt in any way.